src/module-vsbridge-indexer-core/Index/IndexOperations.php

```diff
@@ -295,4 +321,140 @@ private function resolveClient($storeId): ClientInterface
     {
         return $this->clientResolver->getClient($storeId);
     }
+
+    /**
+     * @param $storeId
+     */
+    private function checkEsCondition($storeId)
+    {
+        $clusterHealth = $this->checkClustersHealth($storeId);
+        $this->checkMaxBulkQueueRequirement($clusterHealth, $storeId);
+    }
+
+    /**
+     * Check if clusters are in green status
+     *
+     * @param $storeId
+     *
+     * @return array|void
+     */
+    private function checkClustersHealth($storeId)
+    {
+        $clusterHealth = $this->resolveClient($storeId)->getClustersHealth();
+        if ($this->optimizationSettings->checkClusterHealth()) {
+            if ($clusterHealth[0]['status'] !== self::GREEN_HEALTH_STATUS) {
+                throw new \LogicException(
+                    'Can not execute bulk. Cluster health status is ' . $clusterHealth[0]['status']
+                );
+            }
+        }
+
+        return $clusterHealth;
+    }
+
+    /**
+     * Check if pending tasks + batch indexer size (VueStorefrontIndexer indices setting)
+     * are lower than max bulk queue size master node
+     *
+     * @param array $clusterHealth
+     * @param $storeId
+     *
+     * @return void
+     */
+    private function checkMaxBulkQueueRequirement(array $clusterHealth, $storeId): void
+    {
+        if ($this->optimizationSettings->checkMaxBulkQueueRequirement()) {
+            $masterMaxQueueSize = $this->resolveClient($storeId)->getMasterMaxQueueSize();
+            if (
+                $masterMaxQueueSize &&
+                $clusterHealth[0]['pending_tasks'] + $this->getBatchIndexingSize() > $masterMaxQueueSize
+            ) {
+                throw new \LogicException(
+                    'Can not execute bulk. Pending tasks and batch indexing size is greater than max queue size'
```

Comment:
Will this error be logged into our custom log catalog "vsbridge-indexer" ? ( [magento dir]/var/log/vsbridge-indexer/error.log filke)
If not make sure to use "Divante\VsbridgeIndexerCore\Logger\IndexerLogger" and maybe add a new error handler?

Reply:
You are right. I fixed that. Tell me if you think that i should put more information in log file.